### PR TITLE
implemented TeeApplicationsSource

### PR DESCRIPTION
**Pull request by plata**
Merge applications from git and ${application.user.repository}.

fixes: Implement TeeApplicationsSource #332 

This should also have a unit test. However, the complete unit test setup does not yet exist for phoenicis-apps and I didn't want to add it with this pull request.

### CODE REVIEW

File: phoenicis-javafx/src/main/java/com/playonlinux/javafx/controller/apps/AppsController.java [hunk 1]
```diff
@@ -1,41 +1,45 @@
...
             this.view.showWait();
-            localAppsManager.fetchInstallableApplications(
+            appsSource.fetchInstallableApplications(
                     this.view::populate,
                     e -> this.view.showFailure()
             );
         });
-        this.view.setOnSelectCategory(categoryDTO -> this.view.populateApps(categoryDTO.getApplications()));
+        this.view.setOnSelectCategory(categoryDTO -> this.view.populateApps(
+                new ArrayList<ApplicationDTO>(categoryDTO.getApplications().values())));
```
Comment (qparis) [inline phoenicis-javafx/src/main/java/com/playonlinux/javafx/controller/apps/AppsController.java]: Please use the diamond operator here

File: phoenicis-apps/src/main/java/com/playonlinux/apps/dto/CategoryDTO.java [hunk 2]
```diff
@@ -90,7 +90,7 @@ public Builder withName(String name) {
             return this;
         }
 
-        public Builder withApplications(List<ApplicationDTO> applications) {
+        public Builder withApplications(SortedMap<String, ApplicationDTO> applications) {
```
Comment (qparis) [inline phoenicis-apps/src/main/java/com/playonlinux/apps/dto/CategoryDTO.java]: I think we should use the generic Map interface

File: phoenicis-apps/src/main/java/com/playonlinux/apps/TeeApplicationsSource.java [hunk 3]
```diff
@@ -2,20 +2,40 @@
 
 import com.playonlinux.apps.dto.CategoryDTO;
 
-import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
 
 public class TeeApplicationsSource implements ApplicationsSource {
```
Comment (qparis) [inline phoenicis-apps/src/main/java/com/playonlinux/apps/TeeApplicationsSource.java]: Can we write the corresponding unit test?

Comment (plata) [inline phoenicis-apps/src/main/java/com/playonlinux/apps/TeeApplicationsSource.java]: See my pull request message:

> This should also have a unit test. However, the complete unit test setup does not yet exist for phoenicis-apps and I didn't want to add it with this pull request.

Comment (qparis) [inline phoenicis-apps/src/main/java/com/playonlinux/apps/TeeApplicationsSource.java]: All right, but feel free to add it when necessary

File: phoenicis-apps/src/main/java/com/playonlinux/apps/dto/CategoryDTO.java (hunk 2 above)
Comment (plata) [inline phoenicis-apps/src/main/java/com/playonlinux/apps/dto/CategoryDTO.java]: Done.

File: phoenicis-javafx/src/main/java/com/playonlinux/javafx/controller/apps/AppsController.java (hunk 1 above)
Comment (plata) [inline phoenicis-javafx/src/main/java/com/playonlinux/javafx/controller/apps/AppsController.java]: Done.

**Pull request merged**